lib/favorites/index.ts

```diff
@@ -0,0 +1,148 @@
+import type { CreateOptions } from '../types/CreateOptions';
+import type { RequestCallback } from '../types/RequestCallback';
+import type { RequestOptions } from '../types/RequestOptions';
+import type {
+  FavoritesApi,
+  ListFavoritesQueryParameters,
+  ListFavoritesResponse,
+  AddFavoritesBody,
+  AddFavoritesResponse,
+  RemoveFavoriteOptions,
+  RemoveMultipleFavoritesOptions,
+  RemoveFavoritesResponse,
+} from './types';
+import { FavoriteType } from './types';
+import { types } from '../utils/constants';
+
+export function create(options: CreateOptions): FavoritesApi {
+  const requestor = options.requestor;
+
+  const optionsToSend = {
+    url: options.apiUrls.favorites,
+    ...options.clientOptions,
+  };
+
+  const listFavorites = (
+    getOptions: RequestOptions<ListFavoritesQueryParameters, undefined>,
+    callback?: RequestCallback<ListFavoritesResponse>
+  ) => requestor.get({ ...optionsToSend, ...getOptions }, callback);
+
+  const addItemsToFavorites = (
+    postOptions: RequestOptions<undefined, AddFavoritesBody | AddFavoritesBody[]>,
+    callback?: RequestCallback<AddFavoritesResponse>
+  ) => requestor.post({ ...optionsToSend, ...postOptions }, callback);
+
+  const handleFavorites = (
+    postOptions: RequestOptions<undefined, AddFavoritesBody | AddFavoritesBody[]>,
+    callback?: RequestCallback<AddFavoritesResponse>
+  ) => {
+    const body: AddFavoritesBody = {
+      type: (postOptions as any).type,
+      objectId: (postOptions as any).objectId,
+    };
+    const options = { ...postOptions, body };
+    return addItemsToFavorites(options, callback);
+  };
+
+  const buildFavoriteAddition = (type: string) => {
```

Comment:
```suggestion
  const buildFavoriteAddition = (type: FavoriteType) => {
```

This can be safely replaced because `buildFavoriteAddition` is a private function.